test/e2e/proxy_test.go

```diff
@@ -61,10 +65,124 @@ func proxyTestSuite(t *testing.T) {
+				assert.True(t, proxyAuthHeaderInUse, "Proxy-Authorization expected for proxied request")
+			})
+
+			t.Run("NO_PROXY request", func(t *testing.T) {
+				// using an endpoint that appears in the noProxyList const
+				nonProxiedUrl := noProxyList[0]
+				out, err := tc.curlHeadersOnly(addr, nonProxiedUrl)
+				require.NoErrorf(t, err, "unable to get curl response to URL %s", nonProxiedUrl)
+
+				httpProxyVarInUse, noProxyVarInUse, proxyAuthHeaderInUse := verifyCurlOutput(out,
+					clusterProxy.Status.HTTPProxy, clusterProxy.Status.NoProxy)
+
+				assert.False(t, httpProxyVarInUse, "http_proxy not expected in request to no_proxy URL")
+				assert.Truef(t, noProxyVarInUse, "expected no_proxy %s in request", clusterProxy.Status.NoProxy)
+				assert.False(t, proxyAuthHeaderInUse, "Proxy-Authorization header not expected for no_proxy request")
+			})
+		})
+	}
+}
+
+// curlHeadersOnly runs curl through Windows CMD to get all request headers and trace any redirects
+/*
+Example output:
+* Uses proxy env variable no_proxy == '.cluster.local,.svc,10.128.0.0/14,10.94.72.128/25,127.0.0.1,172.30.0.0/16,api-int.xxxxxxxxxxxxxx.vmc-ci.devcluster.openshift.com,google.com,localhost,redhat.io,static.redhat.com'
+* Uses proxy env variable http_proxy == 'http://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@10.94.72.159:3128'
+*   Trying 10.94.72.159:3128...
+* Connected to 10.94.72.159 (10.94.72.159) port 3128 (#0)
+* Proxy auth using Basic with user 'proxy-user1'
+> HEAD http://raw.githubusercontent.com/openshift/windows-machine-config-operator/master/README.md HTTP/1.1
+> Host: raw.githubusercontent.com
+> Proxy-Authorization: Basic XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+> User-Agent: curl/8.0.1
+> Accept: */ /*
+> Proxy-Connection: Keep-Alive
+>
+
+< HTTP/1.1 301 Moved Permanently
+< Content-Length: 0
+< Server: Varnish
+< Retry-After: 0
+< Location: https://raw.githubusercontent.com/openshift/windows-machine-config-operator/master/README.md
+< Accept-Ranges: bytes
+< Date: Wed, 10 Jan 2024 23:06:14 GMT
+< X-Served-By: cache-dfw-kdal2120069-DFW
+< X-Cache: HIT
+< X-Cache-Hits: 0
+< X-Timer: S1704927975.908992,VS0,VE0
+< Access-Control-Allow-Origin: *
+< Cross-Origin-Resource-Policy: cross-origin
+< Expires: Wed, 10 Jan 2024 23:11:14 GMT
+< Vary: Authorization,Accept-Encoding
+< X-Cache: MISS from 951.27.49.01.in-addr.arpa
+< X-Cache-Lookup: MISS from 951.27.49.01.in-addr.arpa:3128
+< Via: 1.1 varnish, 1.1 951.27.49.01.in-addr.arpa (squid/4.4)
+< Connection: keep-alive
+<
+...
+< HTTP/1.1 200 Connection established
+<
+...
+*/
+func (tc *testContext) curlHeadersOnly(nodeAddress, urlToCurl string) (string, error) {
+	command := fmt.Sprintf("cmd.exe /c curl -vsIL %s", urlToCurl)
+	return tc.runPowerShellSSHJob("curl-headers-only", command, nodeAddress)
+}
+
+// verifyCurlOutput parses the output of a curl request searching for any proxy variables in use and proxy headers
+func verifyCurlOutput(output, httpProxyURL, noProxyList string) (bool, bool, bool) {
```

Comment:
should this be named noProxyList, or noProxyURL?
if a comma separated list is passed in would that work?